optaplanner-quarkus-integration/optaplanner-quarkus/drools-integration-test/src/main/java/org/optaplanner/quarkus/drools/it/domain/TestdataQuarkusEntity.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.quarkus.drools.it.domain;
```

Comment:
I do not see the need for the `it` subpackage. A domain is a domain, regardless if it's for a test or for an integration test.

Reply:
Also, we use the `testdata` keyword for that, which for some reason you are removing later in this PR.

Reply:
That was because Sonarcloud doesn't like it when two modules share the same package (`org.optaplanner.quarkus.testdata` existed in both `optaplanner-quarkus-integration-test` and `optaplanner-quarkus`, causing a failure). The `it` acts as a prefix to prevent package conflicts.

Reply:
Honestly, I don't think we should mess up our code because of a shortcoming in some downstream system, even if it's as important as Sonar.
What are clean alternatives?

Reply:
TBH, using `it` inside package name for files for integration tests is convention for Quarkus: https://github.com/quarkusio/quarkus/tree/main/integration-tests/resteasy-jackson/src/main/java/io/quarkus/it/resteasy/jackson https://github.com/quarkusio/quarkus/tree/main/integration-tests/redis-client/src/main/java/io/quarkus/redis/it etc.

Reply:
I see your point, and this is where things become a bit hairy. Which convention do we pick? The one which is closer (OP), or the one which is more generic (Quarkus)?

I'm not saying that there is a universal answer here, and reasonable people could disagree. That said, I prefer the convention that is closer.

Reply:
I agree with Lukas. This is optaplanner-quarkus-*, not quarkus-optaplanner-*.
That being said, we should actively look at quarkus conventions and align with them if it makes sense. But here I would just stick with the optaplanner way, it's just a package name.

Reply:
Package name should use "drl" instead of "drools".

Reply:
maybe using "it" instead of "testdata" does make more sense. Especially as we have more integration-test modules and everyhing in that "it" package is a single project to test. But then we need to do it for all it tests, not just this one.
@triceo Wdyt?

Proposal A)
=> org.optaplanner.quarkus.drl.it.domain here
and in "integration-test": org.optaplanner.quarkus.testdata.domain => org.optaplanner.quarkus.it.domain
and similar for the jackson and jsonb integration-tests.

Reply:
I see Chris has made the changes in the other files too (great Chris!). So this PR is actually proposal B).

Thinking more on this, I am +1 to proposal B).